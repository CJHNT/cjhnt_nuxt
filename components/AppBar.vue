<script setup>
const { loggedIn, user, clear, fetch } = useUserSession()
const router = useRouter()

fetch()
const logout = () => {
  clear()
  router.push({ name: 'index' })
}
</script>

<template>
  <v-app-bar color="#027f42">
    <template v-slot:prepend>
      <v-btn class="d-md-none"
        ><v-img
          src="@/assets/img/cjh-logo-quadratisch-klein.png"
          alt="Logo: CJHNT Project"
          width="32"
          inline
        ></v-img
      ></v-btn>
      <v-btn
        icon="mdi-information-outline"
        href="https://cjhnt-info.saw-leipzig.de/de"
        target="_blank"
      ></v-btn>
      <v-btn icon="mdi-home" href="/"></v-btn>
    </template>
    <template v-slot:append>
      <v-tooltip location="bottom" text="Advanced Search">
        <template v-slot:activator="{ props }">
          <v-btn icon="mdi-magnify" to="/search/advanced" v-bind="props"></v-btn>
        </template>
      </v-tooltip>

      <template v-if="loggedIn">
        <v-tooltip location="bottom" :text="`Logged in as ${user.email}`">
          <template v-slot:activator="{ props }">
            <v-btn icon="mdi-account" to="/auth/dashboard" v-bind="props"></v-btn>
            <v-btn icon="mdi-logout" @click="logout" v-bind="props"></v-btn>
          </template>
        </v-tooltip>
      </template>
      <v-btn v-else icon="mdi-login" to="/auth/login"></v-btn>
      <LangSwitcher />
    </template>
  </v-app-bar>
</template>
