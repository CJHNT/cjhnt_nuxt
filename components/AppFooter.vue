<script setup>
const props = defineProps({ border: { type: String, default: '' } })
</script>

<template>
  <v-footer class="d-flex flex-column" :app="true" :border="props.border">
    <v-container class="pa-0">
      <v-row no-gutters justify="center">
        <v-col cols="auto">
          <a href="https://www.akademienunion.de/forschungsprojekte/" target="_blank">
            <v-row>
              <v-col cols="auto">
                <picture>
                  <source
                    media="(max-width: 600px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="20"
                    width="60"
                  >
                  <source
                    media="(max-width: 960px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="30"
                    width="90"
                  >
                  <source
                    media="(max-width: 1280px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="40"
                    width="120"
                  >
                  <source
                    media="(max-width: 1920px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="50"
                    width="150"
                  >
                  <source
                    media="(max-width: 2560px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="60"
                    width="180"
                  >
                  <img
                    src="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    alt="Logo: Akademienunion"
                    height="70"
                    width="210"
                  >
                </picture>
              </v-col>
              <v-col cols="auto">
                <picture>
                  <source
                    media="(max-width: 600px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="20"
                    width="14"
                  >
                  <source
                    media="(max-width: 960px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="30"
                    width="21"
                  >
                  <source
                    media="(max-width: 1280px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="40"
                    width="28"
                  >
                  <source
                    media="(max-width: 1920px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="50"
                    width="35"
                  >
                  <source
                    media="(max-width: 2560px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="60"
                    width="42"
                  >
                  <img
                    src="@/assets/img/sachsen-signet_gruen.png"
                    alt="Logo: Akademienunion"
                    height="70"
                    width="47"
                  >
                </picture>
              </v-col>
              <v-tooltip activator="parent" location="top"
                >Dieses Forschungsvorhaben ist Teil im Akademienprogramm, das als derzeit größtes
                geistes- und kulturwissenschaftliches Langfrist-Forschungsprogramm der
                Bundesrepublik Deutschland von Bund und Ländern getragen wird. Diese Maßnahme wird
                mitfinanziert durch Steuermittel auf der Grundlage des vom Sächsischen Landtag
                beschlossenen Haushalts.</v-tooltip
              >
            </v-row>
          </a>
        </v-col>
      </v-row>
    </v-container>
    <div />
  </v-footer>
</template>

<style scoped>
.text-caption.footer-text {
  line-height: normal;
}
</style>

<style>
.v-tooltip__content {
  pointer-events: initial;
}
</style>
