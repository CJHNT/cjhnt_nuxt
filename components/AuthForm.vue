<script setup>
defineProps({
  loading: { type: Boolean, default: false },
  title: { type: String, required: true },
  label: { type: String, default: '' }
})
const emit = defineEmits(['submit'])
const email = ref('')
const repeatEmail = ref('')
const password = ref('')
const repeatPassword = ref('')
const showPassword = ref(false)
const showRepeatPassword = ref(false)
const wantsUpdates = ref(false)

function validateEmail() {
  const pattern =
    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
  return pattern.test(email.value) || 'Invalid e-mail.'
}

const submit = () => {
  const payload = {
    email: email.value,
    repeatEmail: repeatEmail.value,
    password: password.value,
    repeatPassword: repeatPassword.value,
    wantsUpdates: wantsUpdates.value
  }
  emit('submit', payload)
}
</script>

<template>
  <v-card width="100%" class="mx-auto mt-5">
    <v-form @submit.prevent="submit">
      <v-card-title class="pb-0 text-h5">{{ $t(title) }}</v-card-title>
      <v-card-text>
        <v-text-field
          v-if="['auth.signUp', 'auth.changeEmail', 'auth.login'].includes(title)"
          v-model="email"
          :label="label ? $t(label) : 'Email'"
          :rules="[validateEmail()]"
          prepend-icon="mdi-email"
        />
        <v-text-field
          v-if="title === 'auth.changeEmail'"
          v-model="repeatEmail"
          :label="$t('auth.verifyEmail')"
          prepend-icon="mdi-email"
          :rules="[() => email === repeatEmail || $t('auth.noEmailMatch')]"
        />
        <v-text-field
          v-if="['auth.signUp', 'auth.changePassword', 'auth.login'].includes(title)"
          v-model="password"
          :type="showPassword ? 'text' : 'password'"
          :label="label ? $t(label) : $t('auth.password')"
          prepend-icon="mdi-lock"
          :append-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
          @click:append="showPassword = !showPassword"
        />
        <v-text-field
          v-if="['auth.signUp', 'auth.changePassword'].includes(title)"
          v-model="repeatPassword"
          :type="showRepeatPassword ? 'text' : 'password'"
          :label="$t('auth.verifyPassword')"
          prepend-icon="mdi-lock"
          :append-icon="showRepeatPassword ? 'mdi-eye' : 'mdi-eye-off'"
          :rules="[() => password === repeatPassword || $t('auth.noPasswordMatch')]"
          @click:append="showRepeatPassword = !showRepeatPassword"
        />
        <v-checkbox
          v-if="['auth.signUp'].includes(title)"
          v-model="wantsUpdates"
          :label="label ? $t(label) : $t('auth.sendMeUpdates')"
        />
      </v-card-text>
      <v-divider />
      <v-card-actions>
        <v-btn color="success" :disabled="loading" type="submit">
          <template v-if="loading">{{ $t('wait') }}</template>
          <template v-else-if="title === 'auth.changeUpdateStatus'">{{ $t(label) }}</template>
          <template v-else>{{ $t(title) }}</template>
        </v-btn>
      </v-card-actions>
      <template v-if="['auth.signUp', 'auth.login'].includes(title)">
        <v-divider />
        <v-card-text>
          <template v-if="title === 'auth.signUp'">
            {{ $t('auth.alreadyRegistered') }}
            <nuxt-link :to="{ name: 'auth-login' }">{{ $t('auth.login') }}</nuxt-link>
          </template>
          <template v-else-if="title === 'auth.login'">
            {{ $t('auth.noAccountYet') }}
            <nuxt-link :to="{ name: 'auth-register' }">{{ $t('auth.signUp') }}</nuxt-link>
          </template>
        </v-card-text>
      </template>
    </v-form>
  </v-card>
</template>
