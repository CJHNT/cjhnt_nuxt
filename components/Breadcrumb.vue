<script setup>
const props = defineProps({
  ancestors: { type: Array, required: true },
  index: { type: Number, required: true }
})
const { locale } = useI18n()
</script>

<template>
  <v-breadcrumbs :items="props.ancestors" class="py-0">
    <template v-slot:item="{ item }">
      <span v-if="item.disabled">{{ item.title[locale] }} {{ item.ref }}</span>
      <nuxt-link v-else :to="`/collection/${item.id}`">{{ item.title[locale] }}</nuxt-link>
    </template>
    <template v-slot:prepend>
      <v-icon v-if="index > 0" icon="mdi-subdirectory-arrow-right" size="small"></v-icon>
      <v-icon v-else icon="mdi-slash-forward" size="small"></v-icon>
    </template>
    <template v-slot:divider>
      <v-icon icon="mdi-slash-forward" size="small"></v-icon>
    </template>
  </v-breadcrumbs>
</template>
