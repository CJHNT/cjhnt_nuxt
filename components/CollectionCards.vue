<script setup>
const { locale } = useI18n()
const { data: theTexts } = await useAsyncData('apiAllTexts', async () => {
  const allTexts = await $fetch('http://127.0.0.1:5000/api/dts/collections', {
    query: { id: 'all_texts' }
  })
  const textPromises = allTexts.member.map(async (m) => {
    const { data: textData } = await useFetch('http://127.0.0.1:5000/api/dts/collections', {
      query: { id: m['@id'] }
    })
    const returnObject = {
      id: textData.value['@id'],
      de: textData.value['dts:extensions']['dc:title'].find((e) => e['@language'] === 'deu')[
        '@value'
      ],
      en: textData.value['dts:extensions']['dc:title'].find((e) => e['@language'] === 'eng')[
        '@value'
      ]
    }
    return returnObject
  })
  const finishedPromises = await Promise.all(textPromises)
  return finishedPromises
})
const sortedMembers = computed(() => {
  return [...theTexts.value].sort((a, b) => a.id.localeCompare(b.id))
})
import primaryImage from '@/assets/img/1_primary_texts.png'
import commentaryImage from '@/assets/img/commentary.jpg'

function getImage(urn) {
  if (urn === '1_primary_texts') {
    return primaryImage
  } else if (urn === 'urn:cts:cjhnt:commentary') {
    return commentaryImage
  }
}
</script>

<template>
  <v-row justify="center">
    <v-col v-for="member in sortedMembers" :key="member.id" cols="auto">
      <template>
        <nuxt-link class="mx-auto" :to="`/collection/${member.id}`" :title="member[locale]">
          <picture>
            <source media="(max-width: 600px)" :srcset="getImage(member.id)" height="100" />
            <source media="(max-width: 960px)" :srcset="getImage(member.id)" height="130" />
            <source media="(max-width: 1280px)" :srcset="getImage(member.id)" height="160" />
            <img :src="getImage(member.id)" :alt="member[locale]" height="200" />
          </picture>
        </nuxt-link>
      </template>
    </v-col>
  </v-row>
</template>
