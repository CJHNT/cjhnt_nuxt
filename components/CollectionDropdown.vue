<script setup>
defineProps({
  collId: {
    type: String,
    required: true,
    default: ''
  },
  collName: {
    type: String,
    required: true,
    default: ''
  }
})
</script>

<template>
  <div>
    <button>{{ collName }}</button>
    <p v-if="pending">Loading...</p>
    <CollectionDropdownContent v-else v-for="member in data.member" :key="member['@id']">{{
      member.title
    }}</CollectionDropdownContent>
  </div>
</template>
