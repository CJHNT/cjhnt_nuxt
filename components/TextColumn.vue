<script setup>
import { Xslt, XmlParser } from 'xslt-processor'
const { locale } = useI18n()

const props = defineProps({ urn: String, reff: { type: String, default: 1 } })
const reffDepth = () => {
  const reffSections = props.reff.match(/\./g)
  if (reffSections) {
    return reffSections.length + 1
  }
  return 1
}
const { data: docMeta } = await useFetch('/api/dts/collections', {
  body: { id: props.urn },
  method: 'POST'
})
const docTitle = {
  de: docMeta.value['dts:extensions']['dc:title'].find((e) => e['@language'] === 'deu')
    ? docMeta.value['dts:extensions']['dc:title'].find((e) => e['@language'] === 'deu')['@value']
    : docMeta.value.title,
  en: docMeta.value['dts:extensions']['dc:title'].find((e) => e['@language'] === 'eng')
    ? docMeta.value['dts:extensions']['dc:title'].find((e) => e['@language'] === 'eng')['@value']
    : docMeta.value.title
}
const { data: navReturn } = await useFetch('/api/dts/navigation', {
  body: { id: props.urn, level: reffDepth() },
  method: 'POST'
})
const validReffs = navReturn.value['hydra:member'].map((r) => r.ref)
let usedReff = props.reff
const alertText = ref('')
if (!validReffs.includes(props.reff)) {
  alertText.value = `Reference ${usedReff} not found in ${props.urn}. Returning the text's first ${navReturn.value.citeType} (${validReffs[0]}).`
  usedReff = validReffs[0]
}

const currentIndex = validReffs.findIndex((m) => m === usedReff)
const prevId = currentIndex > 0 ? validReffs[currentIndex - 1] : null
const nextId = currentIndex + 1 < validReffs.length ? validReffs[currentIndex + 1] : null

const { data: xmlText } = await useFetch('/api/dts/document', {
  body: { id: props.urn, ref: usedReff },
  method: 'POST'
})
var rawXsl = ''
if (props.urn.includes('commentary')) {
  rawXsl = (await import('../assets/source/commentary.xsl?raw')).default
} else if (
  props.urn.includes('tlg0031') ||
  props.urn.includes('tlg0527') ||
  props.urn.includes('1henoch')
) {
  rawXsl = (await import('../assets/source/nt_fragment.xsl?raw')).default
} else if (props.urn.includes('qumran')) {
  rawXsl = (await import('../assets/source/qumran.xsl?raw')).default
} else {
  rawXsl = (await import('../assets/source/epidoc.xsl?raw')).default
}
const xsltClass = new Xslt()
const xmlParser = new XmlParser()
const parsedXslt = xmlParser.xmlParse(rawXsl)
const parsedText = await xsltClass.xsltProcess(xmlParser.xmlParse(xmlText.value), parsedXslt)
const formattedText = computed(() => {
  return parsedText.replaceAll('span><span', 'span> <span')
})
</script>

<template>
  <v-container>
    <v-row justify="center">
      <v-col v-if="props.urn.includes('qumran')" cols="4">
        <QumranZeichenerklÃ¤rung />
      </v-col>
      <v-col cols="8">
        <v-container>
          <v-row v-if="alertText">
            <v-alert closable density="compact" type="warning">{{ alertText }}</v-alert>
          </v-row>
          <v-row justify="center">
            <v-col cols="auto" class="pr-0">
              <h1>{{ docTitle[locale] }} {{ usedReff }}</h1>
            </v-col>
            <v-col v-if="docMeta['dts:extensions']['dc:rights']" cols="auto" class="pl-0">
              <v-dialog max-width="500">
                <template v-slot:activator="{ props: activatorProps }">
                  <v-btn v-bind="activatorProps" variant="plain" size="x-small"
                    ><v-icon> mdi-copyright </v-icon></v-btn
                  >
                </template>

                <template v-slot:default="{ isActive }">
                  <v-card title="Copyright">
                    <v-card-text>
                      {{ docMeta['dts:extensions']['dc:rights'] }}
                    </v-card-text>

                    <v-card-actions>
                      <v-spacer></v-spacer>

                      <v-btn :text="$t('closeDialog')" @click="isActive.value = false"></v-btn>
                    </v-card-actions>
                  </v-card>
                </template>
              </v-dialog>
            </v-col>
          </v-row>
          <v-row v-if="formattedText">
            <v-col cols="auto">
              <NuxtLink :to="`/texts/${props.urn};${prevId}`" v-show="prevId !== null">{{
                $t('comptext.previous')
              }}</NuxtLink>
            </v-col>
            <v-col>
              <ToReffDropdown
                :currentReff="usedReff"
                :allReffs="validReffs"
                :reffName="navReturn.citeType"
                :textUrn="props.urn"
              />
            </v-col>
            <v-col cols="auto">
              <NuxtLink :to="`/texts/${props.urn};${nextId}`" v-show="nextId !== null">{{
                $t('comptext.next')
              }}</NuxtLink>
            </v-col>
          </v-row>
        </v-container>
        <v-container class="text-content" v-html="formattedText"></v-container>
      </v-col>
    </v-row>
  </v-container>
</template>

<style></style>
