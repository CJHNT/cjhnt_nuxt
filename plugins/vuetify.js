import '@/styles/main.scss'
export default defineNuxtPlugin(() => {})
