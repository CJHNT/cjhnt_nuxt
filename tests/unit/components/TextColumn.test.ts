import { mountSuspended } from '@nuxt/test-utils/runtime'
import TextColumn from '~/components/TextColumn.vue'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'
import { createI18n } from 'vue-i18n'
import enLang from '~/lang/en.json'
import deLang from '~/lang/de.json'

const vuetify = createVuetify({
  components,
  directives
})

const i18n = createI18n({
  locale: 'en',
  messages: {
    en: enLang,
    de: deLang
  }
})

describe('Commentary Column Component', async () => {
  const enWrapper = await mountSuspended(TextColumn, {
    props: { urn: 'urn:cts:greekLit:tlg0018.tlg001.1st1K-eng1', reff: 'title', index: 0 },
    global: {
      plugins: [vuetify, i18n]
    },
    route: '/texts/urn:cts:greekLit:tlg0018.tlg001.1st1K-eng1;title'
  })
  test('component mounts', () => {
    expect(enWrapper.html()).toMatchSnapshot()
  })
  // I think this doesn't work because the textPage.js file is not loaded on the component but on the page
  // test('clicking belegstelle-erlautert reveals the text in section D', async () => {
  //   const clickElement = enWrapper.find('[class*=belegstelle-erl]')
  //   const targetEl = enWrapper.find(`${clickElement.attributes('data-target')}`)
  //   let expandElement = targetEl.element.parentElement
  //   expect(expandElement?.classList.contains('border-opacity-0')).toBe(true)
  //   await clickElement.trigger('click')
  //   expandElement = targetEl.element.parentElement
  //   expect(expandElement?.classList.contains('border-opacity-0')).toBe(false)
  // })
})
