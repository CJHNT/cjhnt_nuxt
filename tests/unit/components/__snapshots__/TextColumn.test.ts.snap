// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Commentary Column Component > component mounts 1`] = `
"<div class="v-container v-locale--is-ltr" index="0">
  <div class="v-row justify-center">
    <!--v-if-->
    <div class="v-col v-col-12">
      <div class="v-container v-locale--is-ltr">
        <!--v-if-->
        <div class="v-row justify-center">
          <div class="v-col v-col-auto pr-0">
            <h1>De Opificio Mundi (eng) title</h1>
          </div>
          <div class="v-col v-col-auto pl-0"><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-x-small v-btn--variant-plain" aria-haspopup="dialog" aria-expanded="false"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-copyright mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
              <!---->
              <!---->
            </button>
            <!---->
          </div>
        </div>
        <div class="v-row justify-space-between">
          <div class="v-col"><a href="/texts/urn:cts:greekLit:tlg0018.tlg001.1st1K-eng1;null" style="display: none;">Previous</a></div>
          <div class="v-col"><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-x-small v-btn--variant-text" aria-haspopup="menu" aria-expanded="false" aria-owns="v-menu-1"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator="">Go to</span>
              <!---->
              <!---->
            </button>
            <!---->
          </div>
          <div class="v-col"><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-x-small v-btn--variant-text" aria-haspopup="menu" aria-expanded="false" aria-owns="v-menu-3"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator="">Read in</span>
              <!---->
              <!---->
            </button>
            <!---->
          </div>
          <div class="v-col"><a href="/texts/urn:cts:greekLit:tlg0018.tlg001.1st1K-eng1;1">Next</a></div>
        </div>
      </div>
      <div class="v-container v-locale--is-ltr text-content">
        <div class="v-row">
          <div class="v-col">
            <!-- eslint-disable-next-line vue/no-v-html -->
            <div>
              <div class="citation-section" n="title" id="urn:cts:greekLit:tlg0018.tlg001.1st1K-eng1;title"><span class="text-header">

      <span lemma="" n="w-1">A</span>
                <span lemma="" n="w-2">TREATISE</span>
                <span lemma="" n="w-3">ON</span>
                <span lemma="" n="w-4">THE</span>
                <span lemma="" n="w-5">ACCOUNT</span>
                <span lemma="" n="w-6">OF</span>
                <span lemma="" n="w-7">THE</span>
                <span lemma="" n="w-8">CREATION</span>
                <span lemma="" n="w-9">OF</span>
                <span lemma="" n="w-10">THE</span>
                <span lemma="" n="w-11">WORLD</span>
                ,
                <span lemma="" n="w-12">AS</span>
                <span lemma="" n="w-13">GIVEN</span>
                <span lemma="" n="w-14">BY</span>
                <span lemma="" n="w-15">MOSES</span>
                .

                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
